modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.module

```diff
@@ -433,11 +433,9 @@ function social_group_flexible_group_social_group_join_method_usage(): array {
  * Implements hook_social_tagging_type_alter().
  */
 function social_group_flexible_group_social_tagging_type_alter(array &$items): void {
-  $items['group']['sets'][] = [
-    'bundles' => ['flexible_group'],
-    'group' => 'additional_details',
-    'label' => t('Tags'),
-  ];
+  $items['group']['sets'][0]['bundles'][] = 'flexible_group';
+  $items['group']['sets'][0]['group'] = 'additional_details';
+  $items['group']['sets'][0]['label'] = t('Tags');
```

Comment:
Nice idea, this matches with now we do this with nodes and bundles.

Although this also means that now our settings change;
If you look at how we save our taggings and load them in `SocialTaggingSettingsForm` it's based on the actual field and formstate values.
Because this hook is also invoked for that form, it also means that we're not backwards compatible anymore.

You can see that here in SocialTaggingService:547

```
          // Filter entities that are not enabled for the platform.
          if (!empty($settings->get("tag_{$entity_type}_type_$bundle"))) {
            // Key now contains type and bundle.
            $options[$entity_type . '_' . $bundle] = $title;
          }
```
We're now going from
`$settings->get("tag_type_group")`
to
`$settings->get("tag_group_type_flexible_group")`

So all of a sudden our settings are not compatible anymore.

You can test this out by going to main, enabling `social_tagging` configuring it so it is available for Group, and add some tagging terms connected to Groups.
Checkout your branch, see that the form seems to be okay, but all of a sudden it's dissappeared as option for the taxonomies.

![Screenshot 2024-07-25 at 10 20 15](https://github.com/user-attachments/assets/3ce257d3-1b0e-4321-9e47-0b9976c9f5ba)

So we need to make sure that the data for the settings also matches the new bundle format in order to be compatible for those that already have this enabled currently for Groups.

Reply:
You are right! Also, I've changed the function `groupActive` of `SocialTaggingServiceInterface`, because we should hide the `SocialGroupTagsBlock` block by group bundle.

Reply:
Cheers @rochek03, interesting.

I think we still have some edge cases we need to worry about.

So right now, we didn't migrate anything from the database, which means if the current scenario is:

![Screenshot 2024-07-29 at 12 46 10](https://github.com/user-attachments/assets/18732efa-dad6-44b4-9db2-15d1402d3f21)

the database for `social_tagging.settings` will be like:

```
Array (
    [enable_content_tagging] => 1
    [allow_category_split] => 1
    [use_category_parent] => 1
    [tag_node_type_book] => 
    [tag_node_type_event] => 1
    [tag_node_type_landing_page] => 
    [tag_node_type_page] => 1
    [tag_node_type_topic] => 1
    [use_and_condition] => 0
    [tag_type_profile] => 0
    [tag_type_group] => 1
)
```

If we update to this code, and the clients looks again he'll see the following settings form:

![Screenshot 2024-07-29 at 12 50 03](https://github.com/user-attachments/assets/a942907d-42e2-4a7d-8fea-30cc0904a6a7)

If he now saves, no changes, the settings will be:

```
Array (
    [enable_content_tagging] => 1
    [allow_category_split] => 1
    [use_category_parent] => 1
    [tag_node_type_book] => 
    [tag_node_type_event] => 1
    [tag_node_type_landing_page] => 
    [tag_node_type_page] => 1
    [tag_node_type_topic] => 1
    [use_and_condition] => 0
    [tag_type_profile] => 0
    [tag_type_group] => 1
    [tag_group_type_flexible_group] => 1
)
```

So we're left with a scenario where `tag_type_group` is true, and there is no way for the customer to make that false.
While in your code you're checking for that scenario. So even if the user disables the flexible group, the tag_type_group is still enabled. That could cause some weird scenario's.

Reply:
@ronaldtebrake I added `hook_update` to delete old config value `tag_type_group`

Reply:
@bramtenhove thanks for the review! I got it, so I changed `hook_update` - I moved the value of `tag_type_group ` to the `tag_group_type_flexible_group` key and deleted the old key `tag_type_group` so existed config will be updated and the value will be saved.